src/ai/backend/common/lock.py

```diff
@@ -76,3 +80,31 @@ def _unlock():
         loop = asyncio.get_running_loop()
         await loop.run_in_executor(self._executor, _unlock)
         return None
+
+
+class EtcdLock(AbstractDistributedLock):
+
+    _con_mgr: EtcdConnectionManager
+    _debug: bool
+    default_timeout: float = 3  # not allow infinite timeout for safety
+
+    def __init__(
+        self, lock_name: str, etcd: AsyncEtcd,
+        *,
+        timeout: Optional[float] = None,
+        debug: bool = False):
```

Comment:
Just a minor style fix:
```suggestion
    def __init__(
        self,
        lock_name: str,
        etcd: AsyncEtcd,
        *,
        timeout: Optional[float] = None,
        debug: bool = False,
) -> None:
```